### ISSUE TITLE
[BUG] Docker version flipflopping between 19.03.9 and 20.10.13

### ISSUE THREAD

**Issue by templela** [GoogleCloudPlatform/cloud-builders#828]
## Affected builder image

`gcr.io/cloud-builders/docker`

## Expected Behavior

Latest should either be pinned to 20 or 19. The codebase has it pointing at 20, but it has clearly been reverted to 19 in the container registry: https://console.cloud.google.com/gcr/images/cloud-builders/GLOBAL/docker 

## Actual Behavior

Latest is pointing at 19, 

## Steps to Reproduce the Problem

N/A

## Additional Info

Either is fine but the repository should be the source of truth, something has been manually reported. 

We're now pinning all our versions just be safe that we don't have this regression. A weekend of missing tags was the result 🤦, our faul.t

**Comment by sgilbert** [comment]
I am struggling with this same issue. It is causing intermittent failures.

**Comment by sutoh** [comment]
I faced a same problem. I have fixed the tag from latest to 20. But I hope 20 is latest.

**Comment by EWhitman09** [comment]
Yep, woke up to broken pipelines as we utilize --all-tags flag which does not work in v19. 'Unkown flag'

Temporary fix is to modify cloudbuild.yaml docker steps across the board to 

`- name: 'gcr.io/cloud-builders/docker:20.10.13'`

**Comment by i2TIC** [comment]
Hello,

I suggest to tag the latest v19 as "19-latest", and latest v20 to "20-latest". Then you'll be able to pin to the latest version of the mayor you need. Also will be nice to have as the real latest the v20.

Best regards

**Comment by bendory** [comment]
The rollout of `20.10.x` turned out to be bumpy due to a change in `docker push` default behavior [issue](https://github.com/docker/cli/pull/2220) that rolled out as part of `20.10.0` [release notes](https://docs.docker.com/engine/release-notes/#client-8).

This is moving forward.
